src/components/Menu/MenuItem/MenuItem.jsx

```diff
@@ -181,6 +182,14 @@ const MenuItem = forwardRef(
     }, [iconBackgroundColor, disabled]);
 
     const renderMenuItemIconIfNeeded = () => {
+      if (iconComponent) {
```

Comment:
not a fan of it = this is a custom renderer for the menu item, if you need one, please create a `CustomMenuItem`

Reply:
Maybe let's not allow all custom components, but make only Avatar component possible here as well as Icon? Like add `avatarIconProps` to the MenuItem? And we have full control to limit avatarProps in case
 I've seen examples of Avatar usage in order to just display a picture - so we'll allow pictures in addition to icons. Seems reasonable to me. What do you think?

Reply:
people are abusing the icon already to pass whatever renderer that they want, if you will allow another one we can't upgrade, change the component without the risk of breaking our users.

The MenuIcon should not be aware of the Avatar
If you want to be on the safe side you could add a AvatarMenuItem and use it with specific props